### ISSUE TITLE
Compatibility with Arduino Zero boards

### ISSUE THREAD

**Issue by trlafleur** [prampec/arduino-softtimer#14]
In testing with a Zero 32bit processor I get errors like this...
So it appears that this fine library is yet NOT compatible with this processor.

```
arduino-softtimer-master/src/BlinkTask.cpp:38:17: error: cannot convert 'volatile uint32_t* {aka volatile long unsigned int*}' to 'volatile uint8_t* {aka volatile unsigned char*}' in assignment
   _portRegister =  portOutputRegister(digitalPinToPort(outPin));
```

**Comment by trlafleur** [comment]
fyi...

I was able to make it run on a zero by changing all of the below statement to int32_t in a number of the .h files.   (This might be fixed with just a cast)
I have only done a very limited test... this may not be the correct long term fix as its just for the Zero M0 processor  

```
//volatile uint8_t *_portRegister;
volatile uint32_t *_portRegister;
```

And for now, I also needed to remove the function of Debouncer, and Rotary that call pciManager from the library

**Comment by TheeCaptain** [comment]
I am using the Feather Huzzah (ESP8266 with some extras) and getting the same problem.

```
SoftTimer\src\BlinkTask.cpp:38:17: error: cannot convert 'volatile uint32_t* {aka volatile unsigned int*}' to 'volatile uint8_t* {aka volatile unsigned char*}' in assignment

   _portRegister = portOutputRegister(digitalPinToPort(outPin));
```

**Comment by prampec** [comment]
Fixed in version 3.1.1.
Please also update the PciManager library (2.1.3 or later) from the LibraryManager.
Please also note, that PCI related options are disabled on a hardware, where PCINT vect. not supported.

**Comment by jishindev** [comment]
Have this same issue with ESP32.

```
BlinkTask.cpp: In member function 'void BlinkTask::init(byte, long unsigned int, long unsigned int, byte, long unsigned int)':
BlinkTask.cpp:42:23: error: cannot convert 'volatile uint32_t* {aka volatile unsigned int*}' to 'volatile uint8_t* {aka volatile unsigned char*}' in assignment
his->_portRegister = portOutputRegister(digitalPinToPort(outPin));
                       ^
```

**Comment by Gixy31** [comment]
Any fix for ESP32? Same here with PCIManager 2.1.4 and SoftTimer 3.2.0

**Comment by Gixy31** [comment]
Solved with this in some .h files 👍 
#if (!defined ESP8266 && !defined ESP32) 
    volatile uint8_t *_portRegister;
#else
    volatile uint32_t *_portRegister;
#endif
